packages/client/src/components/SafeContacts.js

```diff
@@ -1,43 +1,191 @@
-import React from "react";
-import { Person } from "./Svg";
-import { useClipboard } from "../hooks";
+import React, { useState } from "react";
+import { isEmpty } from "lodash";
+import ModalHeader from "./ModalHeader";
+import InputAddress from "./InputAddress";
+import InputText from "./InputText";
+import { Copy, Plus } from "./Svg";
+import { useClipboard, useContacts } from "../hooks";
+import { useModalContext } from "contexts";
 
-function SafeContacts({ safeOwners = [] }) {
+function EditContactModal({ contact, onConfirm, confirmText, headerTitle }) {
+  const modalContext = useModalContext();
```

Comment:
modals should be opened and closed from the outside, suggest passing the closeModal function as a callback instead of bringing  in the modalContext inside of the modal itself